packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -3260,6 +3260,45 @@ void main() {
     final Offset menuTopLeft = tester.getTopLeft(find.bySemanticsLabel('Popup menu'));
     expect(childBottomLeft, menuTopLeft);
   });
+
+  testWidgets('PopupmenuItem onTap should be calling after Navigator.pop', (WidgetTester tester) async {
+    final List<int> orderList = <int>[];
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(
+            actions: <Widget>[
+              PopupMenuButton<int>(
+                onSelected: (int value) {
+                  orderList.add(value);
+                },
+                itemBuilder: (BuildContext context) => <PopupMenuItem<int>>[
+                  PopupMenuItem<int>(
+                    onTap: () {
+                      orderList.add(1);
```

Comment:
Why not make this test to show a modalbottomsheet and make sure it is correctly showed. That will be a more accurate test for the issue this pr is trying to solve.

Reply:
You right. I will change that.